eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/boot/AbstractTCPServer.java

```diff
@@ -0,0 +1,498 @@
+                @Override
+                protected long calculateSize(final Object msg) {
+                    return 1;
+                }
+            };
+            handler.setMaxTimeWait(3_000);
+            return handler;
+        }
+
+    }
+
+    @Sharable
+    private class TcpDispatcher extends SimpleChannelInboundHandler<Package> {
+
+        private final Logger messageLogger = LoggerFactory.getLogger(EventMeshConstants.MESSAGE);
+
+        @Override
+        protected void channelRead0(ChannelHandlerContext ctx, Package pkg) throws Exception {
+            long startTime = System.currentTimeMillis();
+            validateMsg(pkg);
+
+            eventMeshTcpMonitor.getTcpSummaryMetrics().getClient2eventMeshMsgNum().incrementAndGet();
+
+            Command cmd = pkg.getHeader().getCmd();
+            try {
+                if (isNeedTrace(cmd)) {
+                    pkg.getHeader().getProperties()
+                            .put(EventMeshConstants.REQ_C2EVENTMESH_TIMESTAMP, startTime);
+                    pkg.getHeader().getProperties().put(EventMeshConstants.REQ_SEND_EVENTMESH_IP,
+                            eventMeshTCPConfiguration.getEventMeshServerIp());
+                    Session session = clientSessionGroupMapping.getSession(ctx);
+
+                    pkg.getHeader().getProperties().put(EventMeshConstants.REQ_SYS, session.getClient().getSubsystem());
+                    pkg.getHeader().getProperties().put(EventMeshConstants.REQ_IP, session.getClient().getHost());
+                    pkg.getHeader().getProperties().put(EventMeshConstants.REQ_IDC, session.getClient().getIdc());
+                    pkg.getHeader().getProperties().put(EventMeshConstants.REQ_GROUP, session.getClient().getGroup());
+                }
+
+                if (Command.HELLO_REQUEST == cmd || Command.RECOMMEND_REQUEST == cmd) {
+                    if (messageLogger.isInfoEnabled()) {
+                        messageLogger.info("pkg|c2eventMesh|cmd={}|pkg={}", cmd, pkg);
+                    }
+                    processHttpCommandRequest(pkg, ctx, startTime, cmd);
+                    return;
+                }
+
+                if (clientSessionGroupMapping.getSession(ctx) == null) {
+                    if (messageLogger.isInfoEnabled()) {
+                        messageLogger.info("pkg|c2eventMesh|cmd={}|pkg={},no session is found", cmd, pkg);
+                    }
+                    throw new Exception("no session is found");
+                }
+
+                logMessageFlow(ctx, pkg, cmd);
+
+                if (clientSessionGroupMapping.getSession(ctx)
+                        .getSessionState() == SessionState.CLOSED) {
+                    throw new Exception(
+                            "this eventMesh tcp session will be closed, may be reboot or version change!");
+                }
+
+                processHttpCommandRequest(pkg, ctx, startTime, cmd);
```

Comment:
The method name should be changed to `processTcpCommandRequest` would be better . @harshithasudhakar